### PR TITLE
KVStore: Get addressed engine-addr from proxy

### LINKED ISSUE #8382: Introduce two stages FAP

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #8382

Problem Summary:

### What is changed and how it works?

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by CalvinNeo** [comment]
image
img_v3_02cc_bb025186-24b1-4388-9f1e-789b95b334fg
img_v3_02cc_db961c38-398f-4467-a767-301e14ead85g
img_v3_02cd_62f996ea-fa31-4f60-bd98-a6b0b1e7fd8g

File: dbms/src/Storages/KVStore/TMTContext.cpp [hunk 1]
```diff
@@ -170,7 +170,25 @@ TMTContext::TMTContext(
     if (!raft_config.pd_addrs.empty() && S3::ClientFactory::instance().isEnabled()
         && !context.getSharedContextDisagg()->isDisaggregatedComputeMode())
     {
-        s3gc_owner = OwnerManager::createS3GCOwner(context, /*id*/ raft_config.advertise_engine_addr, etcd_client);
+        if (kvstore->getProxyConfigSummay().valid)
+        {
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/TMTContext.cpp:174]: Seems the proxy config summary is only set as valid after `KVStore::restore -> KVStore::fetchProxyConfig`. And that is done after TMTContext instance is build. Can we really get the proxy config to create the S3 GC Owner?

Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/TMTContext.cpp:174]: OK, I will check that.

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/KVStore/TMTContext.cpp [hunk 2]
```diff
@@ -163,14 +163,38 @@ TMTContext::TMTContext(
...
+}
+
+void TMTContext::initS3GCManager(const TiFlashRaftProxyHelper * proxy_helper)
+{
+    kvstore->fetchProxyConfig(proxy_helper);
+    if (!raftproxy_config.pd_addrs.empty() && S3::ClientFactory::instance().isEnabled()
         && !context.getSharedContextDisagg()->isDisaggregatedComputeMode())
     {
-        s3gc_owner = OwnerManager::createS3GCOwner(context, /*id*/ raft_config.advertise_engine_addr, etcd_client);
+        if (kvstore->getProxyConfigSummay().valid)
```
[the remark this answers is not in the document]
Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/TMTContext.cpp:178]: Yes, `fetchProxyConfig` acquires "addressed" proxy config from proxy, which is already set by proxy in `run_impl`.

**Pull request merged**